### PR TITLE
BUG: Set frequency for empty Series

### LINKED ISSUE #14320: BUG: Frequency not set for empty Series
Pandas doesn't set frequency properly for empty series. This is probably related to #14313.
#### A small, complete example of the issue

It works for non-empty series:

``` python
>>> pd.Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]), data=[3]).asfreq('H').index
DatetimeIndex(['2016-09-29 11:00:00'], dtype='datetime64[ns]', freq='H')
```

But for empty series, `freq` is `None`:

``` python
>>> pd.Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H').index
DatetimeIndex([], dtype='datetime64[ns]', freq=None)
```
#### Expected Output

I expected the following output:

``` python
>>> pd.Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H').index
DatetimeIndex([], dtype='datetime64[ns]', freq='H')
```
#### Output of `pd.show_versions()`

<details>
## INSTALLED VERSIONS

commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.7.5-gnu-1
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.19.0rc1
nose: 1.3.6
pip: 8.1.2
setuptools: 27.3.0
Cython: 0.24.1
numpy: 1.11.1
scipy: 0.18.1
statsmodels: None
xarray: None
IPython: 5.0.0
sphinx: 1.4.5
patsy: None
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: None
tables: 3.2.3.1
numexpr: 2.6.1
matplotlib: 1.5.2
openpyxl: None
xlrd: 1.0.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.5.1
html5lib: None
httplib2: 0.9.2
apiclient: 1.5.1
sqlalchemy: 1.0.15
pymysql: None
psycopg2: 2.6.2 (dt dec pq3 ext lo64)
jinja2: 2.8
boto: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by sahildua2305**
- [x] closes #14320 
- [x] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

Alternative to #14340

### CODE REVIEW

File: pandas/tests/frame/test_timeseries.py [hunk 1]
```diff
@@ -323,6 +323,13 @@ def test_asfreq_datetimeindex(self):
         ts = df['A'].asfreq('B')
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        # GH 14340
+        empty = Series(index=pd.DatetimeIndex([])).asfreq('H')
+        normal = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
+                        data=[3]).asfreq('H')
+        self.assertEqual(empty.index.freq, normal.index.freq)
```
Comment (sinhrks) [inline pandas/tests/frame/test_timeseries.py]: can u also test for ``TimedeltaIndex`` and ``PeriodIndex``? (latter looks work properly, but for assurance).

Comment (jreback) [inline pandas/tests/frame/test_timeseries.py]: call these empty -> expected and normal -> result

use ``assert_index_equal`` to compare

Comment (sahildua2305) [inline pandas/tests/frame/test_timeseries.py]: @sinhrks should I add assert it for these types of indices in the same test or should I create separate tests for them?

Comment (sinhrks) [inline pandas/tests/frame/test_timeseries.py]: can u add to ``tserirs/tests/test_period(timedelras)``?

File: doc/source/whatsnew/v0.19.1.txt [hunk 2]
```diff
@@ -80,4 +80,7 @@ Bug Fixes
 
 
 - Bug in ``pd.pivot_table`` may raise ``TypeError`` or ``ValueError`` when ``index`` or ``columns``
-  is not scalar and ``values`` is not specified (:issue:`14380`)
\ No newline at end of file
+  is not scalar and ``values`` is not specified (:issue:`14380`)
+- Bug in ``pd.pivot_table`` may raise ``TypeError`` or ``ValueError`` when ``index`` or ``columns``
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.19.1.txt]: You repeated here the previous entry

File: pandas/tests/frame/test_timeseries.py [hunk 3]
```diff
@@ -323,6 +323,14 @@ def test_asfreq_datetimeindex(self):
         ts = df['A'].asfreq('B')
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        # GH 14340
+        expected = Series(index=pd.DatetimeIndex(
+            ["2016-09-29 11:00"])).asfreq('H')
+        result = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
+                        data=[3]).asfreq('H')
```
Comment (jorisvandenbossche) [inline pandas/tests/frame/test_timeseries.py]: Can you create one of the two without using `asfreq`.

Further, it is not fully clear what is what you actually testing, as the result of the empty series is called `expected` (you typically call the constructed expected value `expected` to compare the result of what you want to test)

**Comment by jorisvandenbossche** [comment]
@sahildua2305 Can you update?

**Comment by sahildua2305** [comment]
Got busy with stuff. I will update this soon @jorisvandenbossche

File: doc/source/whatsnew/v0.19.1.txt (hunk 2 above)
Comment (jreback) [inline doc/source/whatsnew/v0.19.1.txt]: can you move to 0.20.0

**Comment by jreback** [comment]
can you rebase / update

**Comment by sahildua2305** [comment]
Yes, will do!

File: doc/source/whatsnew/v0.19.1.txt (hunk 2 above)
Comment (sahildua2305) [inline doc/source/whatsnew/v0.19.1.txt]: Sure

File: pandas/tests/frame/test_timeseries.py (hunk 3 above)
Comment (sahildua2305) [inline pandas/tests/frame/test_timeseries.py]: @jorisvandenbossche Actually the issue was that we didn't set frequency properly for empty series. Hence I'm testing this particular thing by comparing with frequency of a non-empty `Series` that we make using similar parameters with an empty one.

File: pandas/tests/frame/test_timeseries.py [hunk 4]
```diff
@@ -323,6 +323,14 @@ def test_asfreq_datetimeindex(self):
...
+        self.assert_index_equal(expected.index, result.index)
+
```
Comment (jreback) [inline pandas/tests/frame/test_timeseries.py]: compare the series, e.g. ``assert_series_equal``, and construct the expected directly.

you have two results to compare here, what you currently have labeled as expected and as result.

make another tests where you construct the expected directly (iow, dont' use ``.asfreq``) . IOW you actually should use a direc

File: pandas/tseries/tests/test_period.py [hunk 5]
```diff
@@ -1596,6 +1596,14 @@ def test_asfreq_mult(self):
             self.assertEqual(result.ordinal, expected.ordinal)
             self.assertEqual(result.freq, expected.freq)
 
+    def test_asfreq_period_empty_series(self):
+        # GH 14340
+        expected = Series(index=pd.PeriodIndex(['2011-1', '2011-2', '2011-3'],
```
Comment (jreback) [inline pandas/tseries/tests/test_period.py]: same here as above

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**